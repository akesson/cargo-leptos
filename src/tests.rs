use camino::Utf8PathBuf;

use crate::{ext::path::PathBufExt, run, Cli, Commands, Opts};

#[tokio::test]
async fn workspace_build() {
    let command = Commands::Build(Opts::default());

    let cli = Cli {
        manifest_path: Some(Utf8PathBuf::from("examples/workspace/Cargo.toml")),
        log: Vec::new(),
        command,
    };

    run(cli).await.unwrap();

    let site_dir = Utf8PathBuf::from("target/site");

    insta::assert_display_snapshot!(site_dir.ls_ascii().unwrap_or_default());
}
