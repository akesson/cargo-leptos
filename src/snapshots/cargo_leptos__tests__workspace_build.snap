---
source: src/tests.rs
expression: site_dir.ls_ascii().unwrap_or_default()
---
site:
  project1:
    stuff.txt
    favicon.ico
  project2:
    favicon.ico
    pkg:
      front_package.wasm
      front_package.css.map
      front_package.js
      front_package.css
    pkg:
      project2.css.map
      project2.css
      project2.wasm
      project2.js
